pandas/core/internals.py

```diff
@@ -5686,7 +5707,10 @@ def is_na(self):
     def get_reindexed_values(self, empty_dtype, upcasted_na):
         if upcasted_na is None:
             # No upcasting is necessary
-            fill_value = self.block.fill_value
+
+            # You would think that you want self.block.fill_value here
+            # But in reality that will fill with a bunch of wrong values
+            fill_value = np.nan
```

Comment:
this was surprising to me. If you don't pass in np.nan what ends up happening is that if you merge two sparse frames it will concat with a bunch of `fill_value`'s which is most definitely not what I would expect as a user.

Reply:
huh?

Reply:
Yea this confused me a bunch.

```python
evens = pd.DataFrame({'A': range(0, 20, 2), 'B': range(10)})
threes = pd.DataFrame({'A': range(0, 30, 3), 'B': range(10)})
threes.merge(evens, how="left", on="A")
```

Yields
```
    A  B_x  B_y
0   0    0  0.0
1   3    1  NaN
2   6    2  3.0
3   9    3  NaN
4  12    4  6.0
5  15    5  NaN
6  18    6  9.0
7  21    7  NaN
8  24    8  NaN
9  27    9  NaN
```
As you'd expect for a dense dataframe

but if you sparsify it with a fill_value other than np.nan / None you get weird results

```python
evens = pd.DataFrame({'A': range(0, 20, 2), 'B': range(10)}).to_sparse(fill_value=8675309)
threes = pd.DataFrame({'A': range(0, 30, 3), 'B': range(10)}).to_sparse(fill_value=90210)
threes.merge(evens, how="left", on="A")
```

Yields
```
    A  B_x      B_y
0   0    0        0
1   3    1  8675309
2   6    2        3
3   9    3  8675309
4  12    4        6
5  15    5  8675309
6  18    6        9
7  21    7  8675309
8  24    8  8675309
9  27    9  8675309
```

Is that expected behavior?